sdk/storage/storage-queue/src/QueueClient.ts

```diff
@@ -286,7 +297,7 @@ export class QueueClient extends StorageClient {
     return this.queueContext.create({
       ...options,
       abortSignal: aborter
-    });
+    } as Models.QueueCreateOptionalParams);
```

Comment:
Interesting, I'd thought that this should be compatible with `Models.QueueCreateOptionalParams`. Curious why the cast is needed?

Reply:
I was surprised about this too.  For some reason the TypeScript compiler is comparing the object sent as parameter to the `create()` method to the `ServiceCallback<void>` type instead of `Models.QueueCreateOptionalParams`.  I'd like to figure out the real reason why this is happening so that I can take the type hint back out.  I'll add this to my TODO list.

Reply:
Looking at this a bit more closely, it seems that the `metadata` field of the options type has been changed from an object/dictionary to a string:

https://github.com/Azure/azure-sdk-for-js/pull/3853/files#diff-33e46db3382822f99e85f41eb27fd333L570

Trying to track down why the code generator would change this now.

Reply:
Looks like it has a `string` type in the swagger json? I wonder how the previous code was generated to have a dictionary @XiaoningLiu.

Reply:
I noticed that there's a `#/parameters/Metadata` and a `#/definitions/Metadata` definition where the former has the type set to `string` and the latter is `object` (which seems correct).  Either there's a bug in the Swagger spec or something in the code generator has changed such that it picks up the wrong type.  I'll look into the code generation angle to see if something got broken there.

Reply:
@daviwil Which swagger you are using for the generation? Metadata should definitely be an `object`.

Reply:
Turns out that this was a [bug in `@microsoft.azure/oai2-to-oai3`](https://github.com/Azure/perks/pull/58), I've regenerated the `StorageClient`s after this fix and the correct `metadata` type has now been restored.